puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -40,21 +40,28 @@ def report_to_hash
       resources = build_resources_list
       is_noop = resources.any? { |rs| has_noop_event?(rs) } and resources.none? { |rs| has_failed_event?(rs) }
 
+
+      defaulted_catalog_uuid = defined?(catalog_uuid) ? catalog_uuid : transaction_uuid
```

Comment:
I think this and the same code in catalogs should be refactored to happen in the same place, that way our defaults can't drift apart and we're defaulting in the same way.